signer/kms/aws/go.mod

```diff
@@ -0,0 +1,33 @@
+module github.com/in-toto/go-witness/signer/kms/aws
```

Comment:
I don't believe we want a nested go module here

Reply:
so we can remove the nested go module here. However it's probably worth us referring back to the sigstore folks to ask why the nested go module was implemented here. Maybe they have a reason why they did this 😄. I can't register the kms providers in `kms/signerprovider.go` because it leads to an import cycle, so still need to reference them in `cmd/root.go` of https://github.com/in-toto/witness`.